relay-server/src/actors/project_buffer.rs

```diff
@@ -263,10 +270,51 @@ impl BufferService {
             )))
     }
 
+    /// Saves the provided buffer to the disk.
+    ///
+    /// It returns an error if the spooling failed.
+    async fn do_spool(
+        connection: &mut PoolConnection<Sqlite>,
+        buffer: BTreeMap<QueueKey, Vec<ManagedEnvelope>>,
+    ) -> Result<(), BufferError> {
+        // A builder type for constructing queries at runtime.
+        // This by default creates a prepared sql statement, which is cached and
+        // re-used for sequential queries.
+        let mut query_builder: QueryBuilder<Sqlite> =
+            QueryBuilder::new("INSERT INTO envelopes (own_key, sampling_key, envelope) ");
+
+        // Flatten all the envelopes
+        let envelopes = buffer.into_iter().flat_map(|(k, vals)| {
+            vals.into_iter()
+                .map(move |v| (k, v.into_envelope().to_vec()))
+        });
+
+        // Since we have 3 variables we have to bind, we devide the SQLite limit by 3
+        // here to prepare the chnunks which will be preparing the batch inserts.
+        let mut envelopes = stream::iter(envelopes).chunks(SQLITE_LIMIT_VARIABLE_NUMBER / 3);
+        while let Some(chunk) = envelopes.next().await {
```

Comment:
From offline conversation: Intention here was to not use itertools. However, using some async function on a synchronous iterator from the `futures` crate doesn't buy us anything. Let's use itertools here.